lib/src/discover_tasks.dart

```diff
@@ -66,15 +66,18 @@ class TaskDiscovery {
     }
 
     if (annotation != null) {
-      TaskFunction taskFunction;
+      Function taskFunction;
 
       if (decl is VariableMirror ||
           (decl is MethodMirror && decl.isGetter)) {
         taskFunction = owner.getField(decl.simpleName).reflectee;
       } else if (decl is MethodMirror &&
                  decl.isRegularMethod) {
-        taskFunction = (GrinderContext context) =>
-            owner.invoke(decl.simpleName, [context]);
+        if (decl.parameters.isNotEmpty) {
```

Comment:
Check for at least one positional param to avoid passing a grinder context to something which only has named parameters?